### PR TITLE
[libavif] Update 0.9.1 to 0.9.2

### PR DESCRIPTION
**Pull request by HappySeaFox**
**Describe the pull request**

- #### What does your PR fix?  

Numerous upstream improvements. A couple of patches are not needed anymore.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  

Same as in 0.9.1

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  `Yes`

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  `Yes`

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

File: ports/libavif/fix-assigning-size_t.patch [hunk 1]
```diff
@@ -1,13 +0,0 @@
-diff --git a/src/read.c b/src/read.c
```
Comment (JonLiu1993) [inline ports/libavif/fix-assigning-size_t.patch:1]: @HappySeaFox ,Could you please clarify why this patch should be removed,

Comment (HappySeaFox) [reply ports/libavif/fix-assigning-size_t.patch:1]: Upstream updated the data types, so the cast to `size_t` is not needed anymore.

Comment (JonLiu1993) [reply ports/libavif/fix-assigning-size_t.patch:1]: Thanks for your clarification

**Review by JonLiu1993 — APPROVED** [review]

**Pull request merged**